### DISCUSSION TITLE
Can I use an extension pack to target multiple VS versions?

### DISCUSSION THREAD

**Discussion by chucker** [VsixCommunity/Community.VisualStudio.Toolkit#379] [Q&A]
We have an extension that stopped working with VS 2022 17.2 (specifically, it seems `IVsSolutionBuildManager5` isn't binary-compatible any more), unless we use newer dependencies. Those, in turn, break compatibility with VS 2019.

To avoid having users install entirely different extensions, I'm wondering if I could:

* use MSBuild properties to conditionally reference different dependencies
* write two `vsix`es
* put those in a single [extension pack](https://www.vsixcookbook.com/publish/create-extension-pack.html)

Is that feasible? Is there a simpler way?

**Comment by reduckted** [comment]
I can't answer your question about using an extension pack, but I read recently on the [Visual Studio blog](https://devblogs.microsoft.com/visualstudio/now-introducing-arm64-support-for-vs-extensions/#upload-your-new-extension-versions-on-a-single-marketplace-entry) that the marketplace now supports multiple extensions under a single entry. 

The blog only mentions x64 and ARM, and I haven't tried it, but maybe the marketplace will now let you publish a VS 2019 and VS 2022 extension under the one listing.
